lib/logstash/inputs/twitter.rb

```diff
@@ -79,37 +93,42 @@ def on_body(data)
       c.access_token = @oauth_token
       c.access_token_secret = @oauth_token_secret.value
     end
+
+    @rest = Twitter::REST::Client.new do |c|
+      c.consumer_key = @consumer_key
+      c.consumer_secret = @consumer_secret.value
+      c.access_token = @oauth_token
+      c.access_token_secret = @oauth_token_secret.value
+    end
   end
 
   public
   def run(queue)
     @logger.info("Starting twitter tracking", :keywords => @keywords)
     begin
-      @client.filter(:track => @keywords.join(",")) do |tweet|
-        if tweet.is_a?(Twitter::Tweet)
-          @logger.debug? && @logger.debug("Got tweet", :user => tweet.user.screen_name, :text => tweet.text)
-          if @full_tweet
-            event = LogStash::Event.new(LogStash::Util.stringify_symbols(tweet.to_hash))
-            event.timestamp = LogStash::Timestamp.new(tweet.created_at)
-          else
-            event = LogStash::Event.new(
-              LogStash::Event::TIMESTAMP => LogStash::Timestamp.new(tweet.created_at),
-              "message" => tweet.full_text,
-              "user" => tweet.user.screen_name,
-              "client" => tweet.source,
-              "retweeted" => tweet.retweeted?,
-              "source" => "http://twitter.com/#{tweet.user.screen_name}/status/#{tweet.id}"
-            )
-            event["in-reply-to"] = tweet.in_reply_to_status_id if tweet.reply?
-            unless tweet.urls.empty?
-              event["urls"] = tweet.urls.map(&:expanded_url).map(&:to_s)
-            end
-          end
-
-          decorate(event)
-          queue << event
+      options = {}
+      if @keywords.length > 0
```

Comment:
Once the default value is removed, you can check:

```
if !@keywords.nil? && @keywords.length > 0
```

It complicates the conditions slightly but has the benefit that the documentation doesn't show `Default value is []` that sometimes confuses users.